pkg/hub/deployer/deployer.go

```diff
@@ -172,60 +218,107 @@ func (deployer *Deployer) handleSubscription(subs *appsapi.Subscription) error {
-			return nil
-		}
-
-		chartList, err := deployer.getChartsBySelector(subs, feed)
-		if errors.IsNotFound(err) {
-			msg := fmt.Sprintf("Subscription %s is using a nonexistent HelmChart %s/%s", klog.KObj(subs), subs.Namespace, feed.Name)
-			klog.Error(msg)
-			deployer.recorder.Event(subs, corev1.EventTypeWarning, "NonexistentHelmChart", msg)
-			return nil
-		}
-		if err != nil {
-			msg := fmt.Sprintf("failed to get charts matching %q for Subscription %s: %v", feed, klog.KObj(subs), err)
-			klog.Error(msg)
-			deployer.recorder.Event(subs, corev1.EventTypeWarning, "FailedRetrievingHelmCharts", msg)
-			return err
+		if feed.Kind == helmChartKind.Kind && feed.APIVersion == helmChartKind.Version {
+			chartList, err := deployer.getChartsBySelector(subs, feed)
+			if errors.IsNotFound(err) {
+				msg := fmt.Sprintf("Subscription %s is using a nonexistent HelmChart, feed %s/%s %s/%s %s", klog.KObj(subs), feed.APIVersion, feed.Kind, feed.Namespace, feed.Name, feed.FeedSelector.String())
+				klog.Error(msg)
+				deployer.recorder.Event(subs, corev1.EventTypeWarning, "NonexistentHelmChart", msg)
+				return nil
+			}
+			if err != nil {
+				msg := fmt.Sprintf("failed to get charts matching %q for Subscription %s: %v", feed, klog.KObj(subs), err)
+				klog.Error(msg)
+				deployer.recorder.Event(subs, corev1.EventTypeWarning, "FailedRetrievingHelmCharts", msg)
+				return err
+			}
+			charts = append(charts, chartList...)
+		} else {
+			manifestList, err := deployer.getManifestsBySelector(subs, feed)
+			if errors.IsNotFound(err) {
+				msg := fmt.Sprintf("Subscription %s is using a nonexistent k8s Objects, feed %s/%s %s/%s %s", klog.KObj(subs), feed.APIVersion, feed.Kind, feed.Namespace, feed.Name, feed.FeedSelector.String())
+				klog.Error(msg)
+				deployer.recorder.Event(subs, corev1.EventTypeWarning, "NonexistentK8SObjects", msg)
+				return nil
+			}
+			if err != nil {
+				msg := fmt.Sprintf("failed to get k8s Objects matching %q for Subscription %s: %v", feed, klog.KObj(subs), err)
+				klog.Error(msg)
+				deployer.recorder.Event(subs, corev1.EventTypeWarning, "FailedRetrievingK8SObject", msg)
+				return err
+			}
+			manifests = append(manifests, manifestList...)
 		}
-		charts = append(charts, chartList...)
 	}
 
-	if len(charts) == 0 {
-		deployer.recorder.Event(subs, corev1.EventTypeWarning, "NoHelmCharts", "No helm charts get matched")
+	if len(charts) == 0 && len(manifests) == 0 {
+		deployer.recorder.Event(subs, corev1.EventTypeWarning, "NoMatchedFeeds", "No feeds get matched")
 		return nil
 	}
 
-	// verify HelmChart can be found
-	var chartRefs []appsapi.ChartReference
-	for _, chart := range charts {
-		if len(chart.Status.Phase) == 0 {
-			msg := fmt.Sprintf("HelmChart %s is in verifying", klog.KObj(chart))
-			klog.Warning(msg)
-			deployer.recorder.Event(subs, corev1.EventTypeWarning, "VerifyingHelmChart", msg)
-			return fmt.Errorf(msg)
-		}
+	var allErrs []error
```

Comment:
Should populate to `Base` object here.

And in `handleBase`, we will populate `Description`.

Reply:
will handle HelmChart in another pr later